retrofit/src/main/java/retrofit2/OkHttpCall.java

```diff
@@ -110,16 +110,20 @@
     }
 
     call.enqueue(new okhttp3.Callback() {
-      @Override public void onResponse(okhttp3.Call call, okhttp3.Response rawResponse)
-          throws IOException {
+      @Override public void onResponse(okhttp3.Call call, okhttp3.Response rawResponse) {
         Response<T> response;
         try {
           response = parseResponse(rawResponse);
         } catch (Throwable e) {
           callFailure(e);
           return;
         }
-        callSuccess(response);
+
+        try {
+          callback.onResponse(OkHttpCall.this, response);
+        } catch (Throwable t) {
+          t.printStackTrace();
```

Comment:
this doesn’t seem right

Reply:
oh, but it’s what we’ve always done? Sigh

Reply:
Same as OkHttp. When the callback fails we have nowhere to go.